rp2040-hal/src/xosc.rs

```diff
@@ -107,20 +100,19 @@ impl CrystalOscillator<Disabled> {
         });
 
         //1 ms = 10e-3 sec and Freq = 1/T where T is in seconds so 1ms converts to 1000Hz
-        let delay_to_hz: Hertz = STABLE_DELAY.to_rate().map_err(|_| Error::BadArgument)?;
+        let delay_to_hz: HertzU32 = STABLE_DELAY.try_into_rate().ok_or(Error::BadArgument)?;
 
-        //startup_delay = ((freq_hz * 10e-3) / 256) = ((freq_hz / 1000) / 256)
+        //startup_delay = ((freq_hz * STABLE_DELAY) / 256) = ((freq_hz / delay_to_hz) / 256)
         //See Chapter 2, Section 16, §3)
         //We do the calculation first.
         let startup_delay = frequency
-            .checked_div(&delay_to_hz.integer())
-            .and_then(|r| r.to_generic::<u32>(DIVIDER).ok())
+            .to_Hz()
+            .checked_div(delay_to_hz.to_Hz())
+            .map(|r| r / 256)
             .ok_or(Error::BadArgument)?;
```

Comment:
I think this can be simplified to `let startup_delay = frequency / delay_to_hz / 256;`

Again we know that delay_to_hz is never going to be 0, so checked_div makes less sense...